File: sportsbook-app/wallet-service/src/main/java/com/sportsbook/walletservice/service/WalletService.java
package com.sportsbook.walletservice.service;

import com.sportsbook.walletservice.model.Wallet;
import com.sportsbook.walletservice.repository.WalletRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.math.BigDecimal;

@Service
public class WalletService {

    private final WalletRepository walletRepository;

    public WalletService(WalletRepository walletRepository) {
        this.walletRepository = walletRepository;
    }

    @Transactional
    public void credit(String playerId, BigDecimal amount) {
        Wallet wallet = walletRepository.findById(playerId)
                .orElseGet(() -> new Wallet(playerId, BigDecimal.ZERO));
        wallet.setBalance(wallet.getBalance().add(amount));
        walletRepository.save(wallet);
    }
}

File: sportsbook-app/wallet-service/src/main/kotlin/com/sportsbook/walletservice/service/WalletService.kt
package com.sportsbook.walletservice.service

import com.sportsbook.walletservice.model.Wallet
import com.sportsbook.walletservice.repository.WalletRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.math.BigDecimal

@Service
class WalletService(
    private val walletRepository: WalletRepository,
) {

    @Transactional
    fun credit(playerId: String, amount: BigDecimal) {
        val wallet = walletRepository.findById(playerId)
            .orElseGet { Wallet(playerId, BigDecimal.ZERO) }
        wallet.balance = wallet.balance.add(amount)
        walletRepository.save(wallet)
    }
}
